lib/uploader.js

```diff
@@ -0,0 +1,312 @@
+    const { S3_BUCKET, TESTOMATIO_DISABLE_ARTIFACTS } = this.getConfig();
+    if (!S3_BUCKET) debug(`Upload is disabled because S3_BUCKET is not set`);
+    this.isEnabled = !!(S3_BUCKET && !TESTOMATIO_DISABLE_ARTIFACTS);
+
+    if (this.isEnabled) debug('S3 uploader is enabled');
+    debug(this.getMaskedConfig());
+
+    return this.isEnabled;
+  }
+
+  enableLogStorage() {
+    this.storeEnabled = true;
+  }
+
+  disbleLogStorage() {
+    this.storeEnabled = false;
+  }
+
+  async uploadToS3(Body, Key) {
+    const { S3_BUCKET, TESTOMATIO_PRIVATE_ARTIFACTS } = this.getConfig();
+    const ACL = TESTOMATIO_PRIVATE_ARTIFACTS ? 'private' : 'public-read';
+
+    if (!S3_BUCKET || !Body) {
+      console.log(APP_PREFIX, chalk.bold.red(`Failed uploading '${Key}'. Please check S3 credentials`), this.getMaskedConfig());
+      return;
+    }
+
+    debug('Uploading to S3:', Key);
+
+    const s3 = new S3(this.getS3Config());
+
+    try {
+      const upload = new Upload({
+        client: s3,
+        params: {
+          Bucket: S3_BUCKET,
+          Key,
+          Body,
+          ACL,
+        },
+      });
+
+      const link = await this.getS3LocationLink(upload);
+      this.totalUploaded++;
+      this.succesfulUploads[Key] = link;
+      return link;
+    } catch (e) {
+      this.failedUpload++;
+      debug('S3 uploading error:', e);
+      console.log(APP_PREFIX, "Upload failed:", e.message, this.getMaskedConfig());
+    }
+  }
+
+  readUploadedFiles(runId = null) {
+    const tempFilePath = this.#getUploadFilePath(runId);
+
+    debug('Reading file', tempFilePath);
+
+    if (!fs.existsSync(tempFilePath)) {
+      debug('File not found:', tempFilePath);
+      return [];
+    }
+
+    const stats = fs.statSync(tempFilePath);
+    const diff = (+new Date()) - (+stats.mtime);
+    const diffHours = diff / 1000 / 60 / 60;
+    if (diffHours > 3) {
+      console.log(APP_PREFIX, 'Artifacts file is too old, can\'t process artifacts. Please re-run the tests.');
+      return [];
+    }
+
+    const data = fs.readFileSync(tempFilePath, 'utf8');
+    const lines = data.split('\n').filter(Boolean);
+    return lines.map(line => JSON.parse(line));
+  }
+
+  #getUploadFilePath(runId = null, forceCreate = false) {
+    if (!runId && !forceCreate) {
+      return path.join(os.tmpdir(), 'testomatio.run.latest.jsonl');
+    }
+
+    const tempFilePath = path.join(os.tmpdir(), `testomatio.run.${runId}.jsonl`);
+    if (!fs.existsSync(tempFilePath) || forceCreate) {
+      debug('Creating artifacts file:', tempFilePath);
+      fs.writeFileSync(tempFilePath, '');
+      // make symlink to 'testomatio.run.latest.jsonl' file
+      const latestFilePath = path.join(os.tmpdir(), 'testomatio.run.latest.jsonl');
+      if (fs.existsSync(latestFilePath)) {
+        fs.unlinkSync(latestFilePath);
+      }
+      fs.symlinkSync(tempFilePath, latestFilePath);
+
+    }
+    return tempFilePath;
+  }
+
+  storeUploadedFile(filePath, runId, rid, uploaded = false) {
```

Comment:
maybe `storeUploadedFilePath`?

I would propose to organize params into object `options: {filePath, runId, rid, uploaded}`